src/backend/distributed/utils/foreign_key_relationship.c

```diff
@@ -259,20 +274,10 @@ static void
 GetConnectedListHelper(ForeignConstraintRelationshipNode *node, List **adjacentNodeList,
 					   bool isReferencing)
 {
-	List *neighbourList = NIL;
-
 	node->visited = true;
 
-	if (isReferencing)
-	{
-		neighbourList = node->backAdjacencyList;
-	}
-	else
-	{
-		neighbourList = node->adjacencyList;
-	}
-
 	ForeignConstraintRelationshipNode *neighborNode = NULL;
+	List *neighbourList = GetNeighbourList(node, isReferencing);
 	foreach_ptr(neighborNode, neighbourList)
```

Comment:
I'm tiniest bit uncomfortable by that we use both neighbor and neighbour in this code

Reply:
in this file, #neighbor  >#neighbour, so I will convert neighbour occurences to neighbor